pkgs/applications/misc/clifm/default.nix

```
@@ -0,0 +1,59 @@
+{ stdenv, lib, fetchFromGitHub, makeWrapper, libcap, acl, file, readline
+, withRemoteSupport ? true, sshfs, curlftpfs, cifs-utils
+, withCompressionSupport ? true, atool, archivemount, cdrkit, p7zip, cdrtools
+, withIconSupport ? false, icons-in-terminal ? null
+, extraDeps ? [ ] # for optional plugins' script such as fzf, feh, etc.
+}:
+with lib;
+
+assert withRemoteSupport -> any (p: p != null) [ sshfs curlftpfs cifs-utils ];
+assert withCompressionSupport
+  -> any (p: p != null) [ atool archivemount cdrkit p7zip cdrtools ];
+assert withIconSupport -> icons-in-terminal != null;
+
+let
+  wrapBins = optionals withRemoteSupport [ sshfs curlftpfs cifs-utils ]
+          ++ optionals withCompressionSupport [ atool archivemount cdrkit p7zip
+                                                cdrtools ]
+          ++ optional withIconSupport icons-in-terminal
+          ++ extraDeps;
+in stdenv.mkDerivation rec {
+  pname = "clifm";
+  version = "v1.1";
+
+  src = fetchFromGitHub {
+    owner = "leo-arch";
+    repo = "clifm";
+    rev = version;
+    sha256 = "0mf9lrq0l532vyf4ycsikrw8imn4gkavyn3cr42nhjsr1drygrp8";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  buildInputs = [ libcap acl file readline ];
+
+  makeFlags = [
+    "INSTALLPREFIX=${placeholder "out"}/bin"
+    "DESKTOPPREFIX=${placeholder "out"}/share"
+  ];
+
+  preInstall = ''
+    mkdir -p ${placeholder "out"}/bin
+    mkdir -p ${placeholder "out"}/share
+  '';
+
+  postFixup = ''
+      wrapProgram ${placeholder "out"}/bin/clifm \
+        --prefix PATH : "${makeBinPath wrapBins}"
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = {
+    homepage = "https://github.com/leo-arch/clifm";
+    description =
+      "CliFM is a CLI-based, shell-like, and non-curses terminal file manager written in C: simple, fast, extensible, and lightweight as hell";
```

```suggestion
    description = "CliFM is a CLI-based, shell-like, and non-curses terminal file manager written in C: simple, fast, extensible, and lightweight as hell";
```
maybe we can make this shorter?